dragonfly/implementations/psyllid_provider.py

```diff
@@ -0,0 +1,274 @@
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=0:
+            logger.warning('Deactivating failed')
+            return False
+        else:
+            return True
+
+
+    def reactivate(self, channel):
+        if self.status_value_dict[channel] == 4:
+            logger.info('Reactivating Psyllid instance of channel {}'.format(channel))
+            self.provider.cmd(self.queue_dict[channel], 'reactivate-daq')
+            logger.info(result)
+        else:
+            logger.warning('Cannot reactivate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=4:
+            logger.warning('Reactivating failed')
+            return False
+        else:
+            return True
+
+
+    def quit_psyllid(self, channel):
+        self.provider.cmd(self.queue_dict[channel], 'quit-psyllid')
+        logger.info('psyllid quit!')
+
+
+    @property
+    def all_central_frequencies(self):
+        return self.freq_dict
+
+
+    def get_central_frequency(self, channel):
+        routing_key_map = {
+                           'streaming':'.strw',
+                           'triggered':'.trw'
+                          }
+        request = 'active-config.'+self.channel_dict[channel]+routing_key_map[self.mode_dict[channel]]
+        result = self.provider.get(self.queue_dict[channel]+'.'+request)
+        logger.info('Psyllid says cf is {}'.format(result['center-freq']))
+        self.freq_dict[channel]=result['center-freq']
+        return self.freq_dict[channel]
+
+    
+
+    def set_central_frequency(self, channel, cf):
+        #logger.info('Trying to set cf of channel {} to {}'.format(channel, cf))
+        routing_key_map = {
+                           'streaming':'.strw',
+                           'triggered':'.trw'
+                          }
+        request = '.active-config.'+self.channel_dict[channel]+routing_key_map[self.mode_dict[channel]]
+        try:
+            self.provider.set(self.queue_dict[channel]+request, cf)
+            logger.info('Set central frequency of {} writer for channel {} to {} Hz'.format(self.mode_dict[channel], channel, cf))
+            self.freq_dict[channel]=cf
+            return True
+
+        except core.exceptions.DriplineError:
+            logger.error('Could not set central frequency of Psyllid instance for channel {}'.format(channel))
+            self.freq_dict[channel]=None
+            return False
+        except Exception as e:
+            logger.error('Unexpected error in set_central_frequency')
+            raise 
+
+
+    def start_run(self, channel, duration, filename):
+        payload = {'duration':duration, 'filename':filename}
+        self.provider.cmd(self.queue_dict[channel], 'start-run', payload=payload)
+
+
+    def stop_run(self, channel):
```

Comment:
this method only needs to be called for interrupting a run. I would like to keep this name as long as the command to send to psyllid is stop run as well.